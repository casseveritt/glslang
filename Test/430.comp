#version 430 core

void main()
{
	memoryBarrierAtomicCounter();
	memoryBarrierBuffer();
	memoryBarrierShared();
	memoryBarrierImage();
	groupMemoryBarrier();
}

layout(location = 2) in vec3 v3;
