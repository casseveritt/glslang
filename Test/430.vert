#version 430 core

layout(location = 3) vec4 v4;  // ERROR

layout(location = 4) uniform vec4 uv4;

layout(location = 2) in   inb1 { vec4 v; } b1;  // ERROR
layout(location = 2) out outb1 { vec4 v; } b2;  // ERROR

out gl_PerVertex {
    float gl_ClipDistance[];
};

void foo()
{
    gl_ClipDistance[2] = 3.7;
}

struct sp {
    highp float f;
    in float g;             // ERROR
    uniform float h;        // ERROR
    invariant float i;      // ERROR
    volatile float j;       // ERROR
    layout(row_major) mat3 m3; // ERROR
};

void foo3(invariant vec4 v4,                 // ERROR
          volatile vec3 v3,
          layout(location = 3) vec2 v2,      // ERROR
          centroid vec3 cv3)                 // ERROR
{
}

struct S {
    mat3x2 m[7];  // needs 7*3 locations
    float f;      // needs 1 location
};                // needs 22 locations

layout(location = 10) out S cs[2];     // 10 through 10 + 2 * 22 - 1 = 53
layout(location = 54) out float cf;
layout(location = 53) out float cg; // ERROR, collision at 31

layout(location = 10) in vec4 alias1;
layout(location = 10) in vec4 alias2;  // okay for vertex input on desktop

out float gl_ClipDistance[17];  // ERROR, size too big