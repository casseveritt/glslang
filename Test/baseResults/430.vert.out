430.vert
Warning, version 430 is not yet complete; some version-specific features are present, but many are missing.
ERROR: 0:3: 'location' : can only appy to uniform, buffer, in, or out storage qualifiers 
ERROR: 0:7: 'location qualifier on in/out block' : not supported for this version or the enabled extensions 
ERROR: 0:8: 'location qualifier on in/out block' : not supported for this version or the enabled extensions 
ERROR: 0:21: 'g' : cannot use storage or interpolation qualifiers on structure members 
ERROR: 0:22: 'h' : cannot use storage or interpolation qualifiers on structure members 
ERROR: 0:23: 'i' : cannot use invariant qualifier on structure members 
ERROR: 0:24: 'j' : cannot use memory qualifiers on structure members 
ERROR: 0:25: 'm3' : cannot use layout qualifiers on structure members 
ERROR: 0:28: '' : cannot use invariant qualifier on a function parameter 
ERROR: 0:30: '' : cannot use layout qualifiers on a function parameter 
ERROR: 0:31: '' : cannot use auxiliary or interpolation qualifiers on a function parameter 
ERROR: 0:42: 'location' : overlapping use of location 53
ERROR: 0:47: 'gl_ClipDistance array size' : must be less than gl_MaxClipDistances (8)
ERROR: 13 compilation errors.  No code generated.


ERROR: node is still EOpNull!
0:14  Function Definition: foo( (void)
0:14    Function Parameters: 
0:16    Sequence
0:16      move second child to first child (float)
0:16        direct index (float)
0:16          gl_ClipDistance: direct index for structure (out 17-element array of float)
0:16            '__anon__0' (out block{out 17-element array of float gl_ClipDistance, })
0:16            Constant:
0:16              2 (const uint)
0:16          Constant:
0:16            2 (const int)
0:16        Constant:
0:16          3.700000
0:31  Function Definition: foo3(vf4;vf3;vf2;vf3; (void)
0:31    Function Parameters: 
0:31      'v4' (in 4-component vector of float)
0:31      'v3' (in 3-component vector of float)
0:31      'v2' (in 2-component vector of float)
0:31      'cv3' (in 3-component vector of float)
0:?   Linker Objects
0:?     'v4' (layout(location=3 ) 4-component vector of float)
0:?     'uv4' (layout(location=4 ) uniform 4-component vector of float)
0:?     'b1' (layout(location=2 ) in block{in 4-component vector of float v})
0:?     'b2' (layout(location=2 ) out block{out 4-component vector of float v})
0:?     '__anon__0' (out block{out 17-element array of float gl_ClipDistance, })
0:?     'cs' (layout(location=10 ) smooth out 2-element array of structure{7-element array of 3X2 matrix of float m, float f})
0:?     'cf' (layout(location=54 ) smooth out float)
0:?     'cg' (layout(location=53 ) smooth out float)
0:?     'alias1' (layout(location=10 ) in 4-component vector of float)
0:?     'alias2' (layout(location=10 ) in 4-component vector of float)
0:?     'gl_VertexID' (gl_VertexId int)
0:?     'gl_InstanceID' (gl_InstanceId int)


Linked vertex stage:

ERROR: Linking vertex stage: Missing entry point: Each stage requires one "void main()" entry point


